src/engine/source/cmsync/src/cm.cpp

```diff
@@ -0,0 +1,618 @@
+    }
+
+    // Check if output path exists and is a directory
+    m_coreOutputReader = std::make_unique<CoreOutputReader>(outputPath);
+}
+
+CMSync::~CMSync() = default;
+
+/************************************************************************************
+ * Other public methods or other interfaces can be added here
+ ************************************************************************************/
+
+// Clean catalog
+void CMSync::cleanCatalog(const std::string& ns, const std::vector<api::catalog::Resource::Type>& typesToClean)
+{
+    namespace acns = api::catalog;
+
+    const auto catalog = getHandlerOrThrow(m_catalog, "Catalog");
+
+    // For each type, delete all resources of that type in the namespace
+    for (const auto& type : typesToClean)
+    {
+        acns::Resource resource {acns::Resource::typeToStr(type), acns::Resource::Format::json};
+
+        // Check if exists resources of this type in the namespace
+        if (!catalog->collectionExists(resource, ns))
+        {
+            // No resources to delete
+            continue;
+        }
+
+        // Delete all resources of this type in the namespace
+        const auto error = catalog->deleteResource(resource, ns);
+        if (error)
+        {
+            throw std::runtime_error(fmt::format("Failed to clean resources of type '{}' in namespace '{}': {}",
+                                                 acns::Resource::typeToStr(type),
+                                                 ns,
+                                                 error->message));
+        }
+    }
+}
+
+// Clean KVDB TODO: Separe KVDB between CTI and User (Maybe ns?)
+void CMSync::cleanAllKVDB()
+{
+    auto kvdbManager = getHandlerOrThrow(m_kvdbManager, "KVDB Manager");
+
+    // List all DBs
+    const auto dbs = kvdbManager->listDBs(false);
+
+    // Check if KVDB are in use
+    for (const auto& dbname : dbs)
+    {
+        const auto refCount = kvdbManager->getKVDBHandlersCount(dbname);
+        if (refCount > 0)
+        {
+            throw std::runtime_error(
+                fmt::format("Cannot delete KVDB '{}', it is in use by {} handlers", dbname, refCount));
+        }
+    }
+
+    // Delete all DBs
+    for (const auto& db : dbs)
+    {
+        auto error = kvdbManager->deleteDB(db);
+        if (error)
+        {
+            // TODO: Log error and continue
+            throw std::runtime_error(fmt::format("Failed to delete KVDB '{}': {}", db, error->message));
+        }
+    }
+}
+
+// Remove policys from catalog
+void CMSync::cleanAllPolicys()
+{
+    const auto policyManager = getHandlerOrThrow(m_policyManager, "Policy Manager");
+
+    // List all policys
+    const auto respOrError = policyManager->list();
+    if (base::isError(respOrError))
+    {
+        // TODO: Maybe no are policys
+        throw std::runtime_error(fmt::format("Failed to list policys: {}", base::getError(respOrError).message));
```

Comment:
is there a chance that we could catch this exceptions in the same module to retry a cleanup? can this make any sense?